ibis_substrait/tests/compiler/test_parity.py

```diff
@@ -117,127 +69,164 @@ def run_parity_tests(expr, datasets, compiler, engines=None):
+@pytest.fixture
+def ibis_filter_groupby_count_distinct():
     filter_table = orders.join(
         stores, orders["fk_store_id"] == stores["store_id"]
     ).filter(lambda t: t.order_total > 30)
 
-    grouped_table = filter_table.group_by("city").aggregate(
-        sales=filter_table["city"].nunique()
-    )
-
-    compiler = SubstraitCompiler()
-    run_parity_tests(grouped_table, datasets, compiler=compiler, engines=[])
+    return filter_table.group_by("city").aggregate(sales=filter_table["city"].nunique())
 
 
-def test_aggregate_having():
-    expr = orders.aggregate(
+@pytest.fixture
+def ibis_aggregate_having():
+    return orders.aggregate(
         [orders.order_id.max().name("amax"), orders.order_id.count().name("acount")],
         by="fk_store_id",
         having=(_.order_id.count() > 1),
     )
 
-    compiler = SubstraitCompiler()
-    run_parity_tests(expr, datasets, compiler=compiler)
-
 
-def test_inner_join_chain():
-    expr = orders.join(stores, orders["fk_store_id"] == stores["store_id"]).join(
+@pytest.fixture
+def ibis_inner_join_chain():
+    return orders.join(stores, orders["fk_store_id"] == stores["store_id"]).join(
         customers, orders["fk_customer_id"] == customers["customer_id"]
     )
 
-    compiler = SubstraitCompiler()
-    run_parity_tests(expr, datasets, compiler=compiler)
 
+@pytest.fixture
+def ibis_union():
+    return orders.union(orders)
 
-def test_union():
-    expr = orders.union(orders)
 
-    compiler = SubstraitCompiler()
-    run_parity_tests(expr, datasets, compiler=compiler)
-
-
-# TODO acero doesn't seem to support this, maybe run duckdb on both sides?
-def test_window():
-    expr = orders.select(
+@pytest.fixture
+def ibis_window():
+    return orders.select(
         orders["order_total"].mean().over(ibis.window(group_by="fk_store_id"))
     )
 
-    compiler = SubstraitCompiler()
-    run_parity_tests(expr, datasets, compiler=compiler, engines=[])
 
+@pytest.fixture
+def ibis_is_in():
+    return stores.filter(stores.city.isin(["NY", "LA"]))
+
+
+@pytest.fixture
+def ibis_scalar_subquery():
+    return orders.filter(orders["order_total"] == orders["order_total"].max())
+
+
+all_exprs = [
+    "ibis_projection",
+    "ibis_mutate",
+    "ibis_sort",
+    "ibis_sort_limit",
+    "ibis_filter",
+    "ibis_inner_join",
+    "ibis_left_join",
+    ("ibis_filter_groupby", {"datafusion_consumer": (Exception, "")}),
+    (
+        "ibis_filter_groupby_count_distinct",
+        {
+            "acero_consumer": (pa.ArrowNotImplementedError, "Unimplemented"),
+            "datafusion_consumer": (Exception, ""),
+        },
+    ),
+    ("ibis_aggregate_having", {"datafusion_consumer": (Exception, "")}),
+    "ibis_inner_join_chain",
+    "ibis_union",
+    ("ibis_window", {"acero_consumer": (pa.ArrowNotImplementedError, "Unimplemented")}),
+    "ibis_is_in",
+    (
+        "ibis_scalar_subquery",
+        {"acero_consumer": (pa.ArrowNotImplementedError, "Unimplemented")},
```

Comment:
I think this is probably better handled with `pytest.param` -- so it's likely that we'll need to slightly reorganize how these are being tested.